staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/diskmanagers/vdm.go

```diff
@@ -39,18 +39,30 @@ func (diskManager virtualDiskManager) Create(ctx context.Context, datastore *vcl
 	if diskManager.volumeOptions.SCSIControllerType == "" {
 		diskManager.volumeOptions.SCSIControllerType = vclib.LSILogicControllerType
 	}
-	// Create virtual disk
-	diskFormat := vclib.DiskFormatValidType[diskManager.volumeOptions.DiskFormat]
-	// Create a virtual disk manager
-	vdm := object.NewVirtualDiskManager(datastore.Client())
 	// Create specification for new virtual disk
+	diskFormat := vclib.DiskFormatValidType[diskManager.volumeOptions.DiskFormat]
 	vmDiskSpec := &types.FileBackedVirtualDiskSpec{
 		VirtualDiskSpec: types.VirtualDiskSpec{
 			AdapterType: diskManager.volumeOptions.SCSIControllerType,
 			DiskType:    diskFormat,
 		},
 		CapacityKb: int64(diskManager.volumeOptions.CapacityKB),
 	}
+
+	vdm := object.NewVirtualDiskManager(datastore.Client())
+
+	// Check for existing VMDK before attempting create. Because a name collision
+	// is unlikely, "VMDK already exists" is likely from a previous attempt to
+	// create this volume.
+	infos, err := vdm.QueryVirtualDiskInfo(ctx, diskManager.diskPath, datastore.Datacenter.Datacenter, false)
+	// There is not a stable way to check for NotFound errors, so ignore all errors
```

Comment:
Using `Datastore.Exists` should work as expected here, have you looked at using that instead?
https://github.com/kubernetes/kubernetes/blob/31b7e3d90535aaa83e9125f19f5f933499a1b3be/staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/datastore.go#L106

Reply:
Many thanks for pointing this out! PTAL